### ISSUE TITLE
AtGames Legends Flashback

### ISSUE THREAD

**Issue by kjetil-f** [farmerbb/RED-Project#126]
Using a USB to UART adapter with male to female jumper wire you can dump all of the included games. Even though you can, there is no soldering required to do this.

[See image for reference](https://global.discourse-cdn.com/standard10/uploads/retrogameboards/original/3X/8/d/8d4630f19ca4203fdf233eda14c46af56bfbf4f6.jpeg).

First, format a USB stick to FAT32. (Not sure if it’s necessary, but since it worked on the Super Retro-Cade, I formatted it with the partitioning scheme MBR, using Rufus.)

Use PuTTY (or other terminal programs) and set the serial port to COM (and the correct USB port number. COM5 in my case). Baud rate/speed to 1500000.

When the Linux command prompt starts up type in root and hit enter to get root access. (Not sure why, but I needed to do so before the login screen showed up.).

The games can be found in the folder called rom. and the SD-card inside the folder called media. (/media/usbhd-mmcblk0p1/). You can make a folder at the root of the SD-card called GAMES e.g. and copy the content from the rom-folder into the SD-card: cp /rom/* /media/usbhd-mmcblk0p1/GAMS, or use the cp -r command to copy the whole folder instead (that way you will get the sub-folders copied over as well).

**Comment by kjetil-f** [comment]
![atgameslfb](https://github.com/farmerbb/RED-Project/assets/93976052/2d1995ba-16e0-4187-8d9b-ba485aea02b6)

**Comment by kjetil-f** [comment]
Sorry about that. It's FB8660B actually. Not sure if it matters much for for the purpose of dumping the ROMs though. Seems like as long as the console is Linux based, and have UART-connections, similar methods can be used.

**Comment by RealRelativeEase** [comment]
Thank you for posting the instructions! Guess it's time for me to get one of these as well.

Are the ROMs usable as is when dumped or are they encrypted, like when extracted from firmware updates? And have you been able to confirm whether the controller port 1 issue was a result of accessing the system via UART?

**Comment by kjetil-f** [comment]
I haven't done a thorough test, but the console games seems to be working great. The only issue I found was that the first Super Star Wars wouldn't load past the intro screen when I used an Analogue Pockets openFPGA core. On other software emulators it worked just fine.

The arcade games are a little more hit and miss when I tested them in FinalBurn Neo, but I think I saw references to both MAME 2003 and 2010 in the consoles emulator-folder.

For the controller port issue, it seems to be a design flaw on the hardware level. I've read that several people have this issue without even opening up the console. Seems like it's just a bad connection that are causing it. If you wiggle a bit on the pole with the red power light so it turns on, the controller port will work. So I'm guessing some insulating tape is probably the simples solution for holding it in place. So it's not the UART connection that causing it, but opening up the console might.

**Comment by kjetil-f** [comment]
Good question. There is a couple of folders named **res** and **res_fif**, that contains a texture bmp, control.log, fif.rom and a lot of wav-files. How to make it run, I don't know.

**Comment by kjetil-f** [comment]
This might be helpfull information regarding the Fix it Felix Jr. ROM:
https://github.com/libretro/RetroArch/issues/15821

**Comment by kjetil-f** [comment]
Demo of the AtGames-version of Fix it Felix Jr. to running on an emulator: https://www.youtube.com/watch?v=CjzazRmnIN8&t=50s&ab_channel=TechnoTurnovers

**Comment by kjetil-f** [comment]
Do you power on the device? Is the baud rate sat to 1500000, and the TX wire connected to RX and vice versa?

**Comment by kjetil-f** [comment]
Excellent. Please share if it's anything noteworthy about the verifications. Have never tried to do that.

**Comment by kjetil-f** [comment]
Yeah, me too. I  just Google Linux-commands. The easiest way might be to dump all of the files and use Windows to search for it. But it is possible to search for files in Linux.

Have you by any chance checked the Emulation folder? I was planning research this myself, but been busy lately. Would be interesting to see exactly what emulators (and what versions) that make this thing work.

**Comment by kjetil-f** [comment]
Crazy that Disney allowed it, but they probably wasn't aware. What I find a bit amusing is that since you bought this fair and square, no one can legally take it away from you.

https://bootleggames.fandom.com/wiki/Squirrel_King

**Comment by RealRelativeEase** [comment]
I'm running into a bit of a roadblock with the FB8650-MB-V2. The ports on the motherboard aren't labeled, but I assume they're the same as on the photo. I've got Ground connected to Ground on the USB TTL board, Transmit connected to Receive on the TTL board and vice versa. I've got a 16 GB SD card formatted to FAT32 inside the FB. What is the proper order to access the motherboard in PuTTY? Do I plug the board in before I launch PuTTY? Do I turn on the power before I establish a connection via the TTL board?

**Comment by RealRelativeEase** [comment]
The FB8650 I got shipped with a 128MB SD card with a firmware file named `SD.img` on it. Which I assume includes most of the games, since PuTTY could only transfer about 20 games to a larger external SD card without the firmware update on it. I've examined `SD.img` in binwalk and it seems to include the missing games, but they're all in encrypted .aes format and I assume the FB decrypts them.

Is there a prompt for PuTTY to extract the unencrypted files? Or what would be the prompt to dump the entire FW and external SD content?

**Comment by kjetil-f** [comment]
Think I read somewhere that some models of the FB got a firmware update that you could install from the SD-card, so that's probably it. But would not everything be installed to the console itself if you execute the firmware update?

**Comment by RealRelativeEase** [comment]
I'm not sure if those official firmware updates are installable to begin with. Exposing myself as a total newb here, how would I install the firmware update from the stock SD card? There's no update option when the unit is powered on with the SD card inside.

The serial number on the sticker of my unit also contradicts the serial number on the motherboard.
![P1020225](https://github.com/farmerbb/RED-Project/assets/99961958/74acf28a-732f-435b-b2ce-b0edc05d6413)
![P1020224](https://github.com/farmerbb/RED-Project/assets/99961958/98e8db42-6b23-466c-b65e-0cdd1b1a97c1)

**Comment by RealRelativeEase** [comment]
> It seems you are uncertain what model your unit is. You can tell what model it is by what games it includes: for example, does it have Fix-It Felix? (660 or 660B) Does it have Generals of the Yang Family? (660S) Does it have neither? (650 or 650S)
> 
> Lastly, if you decide to do the update, you need to rename the firmware update to “arcade_update.img” (no quotes, duh).

Fix-It Felix and Generals of the Yang Family are both on `SD.img` on the stock SD card, since they don't show up on the main menu if the SD card is removed from the device when it's powered on. Neither of them is pre-installed on my unit.

**Comment by kjetil-f** [comment]
For what it's worth, my unit the model number on the back says FB8660B, and on the chip it says FB8660MB-V1, 20190329 T=1.0 And Fix-It Felix and Generals of the Yang Family are included on the unit itself. So my Flashback are probably a newer revision then.

**Comment by RealRelativeEase** [comment]
> For what it's worth, my unit the model number on the back says FB8660B, and on the chip it says FB8660MB-V1, 20190329 T=1.0 And Fix-It Felix and Generals of the Yang Family are included on the unit itself. So my Flashback are probably a newer revision then.

The chip on my unit also has a different branding than the regular Rockchip used on these. The Monkey King brand chips seem to have been used in Flashback Mega Drive systems ~2017-2018. My guess is that AtGames (or maybe their European distributor) has been clearing out leftover stock of older Flashback Legends systems, that's why I ended up with a version from 2018, despite being billed as the 2022 version. Yeah, I think I'll ask for a refund.

**Comment by Technoturnovers** [comment]
> Do you know where to find the FIF Libretro core on the Flashback? I suck at using Linux terminals.

Sorry for being over half a year late to this, but since it's been so long and you guys still aren't having any luck finding it, I'm just gonna direct you to search "fif_libretro.so" on archive.org, and you should find what you're looking for

**Comment by psymage** [comment]
@kjetil-f Trying to get into the command prompt but I'm not getting a prompt at all in PuTTy. I tried typing in root as it turns on and pressing enter but all I get is this and I can't get anything to go through at this point. I have the same model as you do: FB8660. My baud rate is set to 1500000 and my wiring appears to be correct.  I wonder if it's my adapter that's the problem.

<img width="1919" height="1044" alt="Image" src="https://github.com/user-attachments/assets/3a0c2f19-bf7f-4830-b195-682cab350866" />

**Comment by kjetil-f** [comment]
Sorry, I'm lost on this one.

**Comment by psymage** [comment]
> Sorry, I'm lost on this one.

Actually just got it to work with a different adapter. I'm in as root now. It must have been the previous adapter not being able to support the baud rate.

**Comment by kjetil-f** [comment]
Glad to hear.
